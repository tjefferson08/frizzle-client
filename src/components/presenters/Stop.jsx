import React from 'react';

const Stop = ({ stop_name, stop_id, stop_distance }) => {
  return (
    <li>
      <h3>{stop_name}</h3>
      <div>
        <em>Id: {stop_id}</em>
      </div>
      <div>
        Distance: {stop_distance} meters
      </div>
    </li>
  );
};

export default Stop;
