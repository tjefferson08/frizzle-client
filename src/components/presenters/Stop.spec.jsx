import React from 'react';
import ReactTestUtils from 'react-addons-test-utils';
import Stop from './Stop';

test('should render a li', () => {
  const renderer = ReactTestUtils.createRenderer();
  const fakeStop = {
    stop_name: 'Stop 1',
    stop_id: '123',
    stop_distance: 1001.1234
  };

  renderer.render(<Stop {...fakeStop} />)
  const result = renderer.getRenderOutput();

  expect(result.type).toBe('li');
  expect(result).toMatchSnapshot();
});

