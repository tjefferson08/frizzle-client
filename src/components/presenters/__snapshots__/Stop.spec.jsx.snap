exports[`test should render a li 1`] = `
<li>
  <h3>
    Stop 1
  </h3>
  <div>
    <em>
      Id: 
      123
    </em>
  </div>
  <div>
    Servicing routes: 
    10, 20, 30
  </div>
  <div>
    Distance: 
    1001.1234
     meters
  </div>
</li>
`;
